### PR TITLE
fix a bug that broke updating records

### PR DESCRIPTION
**Pull request by linki**
When making the migration to using `TXT` records for marking records as "owned" by a particular `mate` instance I slipped in a mistake that prevented DNS records from being updated on Google Cloud DNS. This is a quick workaround. I intend to refactor this file in the future.

/cc @ideahitme @mikkeloscar

### CODE REVIEW

File: consumers/google.go [hunk 1]
```diff
@@ -172,31 +177,40 @@ func (d *googleDNSConsumer) applyChange(change *dns.Change) error {
...
-				records[r.Name] = nil
+		if r.Type == "A" || r.Type == "TXT" {
+			record := records[r.Name]
+
+			if record == nil {
+				record = &ownedRecord{}
 			}
+
+			switch r.Type {
+			case "A":
```
Comment (ideahitme) [inline consumers/google.go:197]: what about CNAME and other types ?

Comment (ideahitme) [reply consumers/google.go:197]: if i am not mistaken, this may result in existing records being overwritten ? (because they are not added to the map)

**Comment by ideahitme** [comment]
we really need some testing for google consumer :P

File: consumers/google.go (hunk 1 above)
Comment (linki) [reply consumers/google.go:197]: Other types are ignored, they won't be touched at all. Only A and TXT records will be touched. I tested those cases manually.

This list is only for removing the records `mate` is responsible for (hence only `A` and `TXT`). These will be removed in the same transaction as the desired records will be added. This avoids filtering for necessary updates in `mate` (worked fine so far).

So by not including other types it means they won't be removed. On the other hand, when adding an A record for the same DNS name without actually being responsible the update will fail as Google has no upsert logic (and the record isn't delete as we are not responsible).

Tested many cases and works as expected but I agree this needs to be redone.

**Comment by linki** [comment]
we definitely do

**Comment by ideahitme** [comment]
👍

**Pull request merged**